.github/workflows/archive-full-index.yaml

```
@@ -0,0 +1,146 @@
+name: Generate the full index archive for Bitnami Charts
+on:
+  schedule:
+    - cron: '0 0 * * *'
+env:
+  LAST_INDEX_URL: https://charts.bitnami.com/bitnami/index.yaml
+  INDEXES_DIR: indexes
+jobs:
+  get:
+    runs-on: ubuntu-latest
+    name: Get
+    steps:
+      - id: checkout-repo
+        name: Checkout repo
+        uses: actions/checkout@v3
+      - id: get-last-indexes
+        name: Get indexes
+        run: |
+          mkdir $INDEXES_DIR
+          curl $LAST_INDEX_URL -o ./$INDEXES_DIR/last_index.yaml
+          cp bitnami/index.yaml ./$INDEXES_DIR/old_index.yaml
+      - id: indexes-lint
+        name: Lint indexes
+        run: |
+          cat << EOF > config
+          extends: relaxed
+
+          rules:
+            indentation:
+              level: error
+            line-length: disable
+          EOF
+          yamllint -c config ./$INDEXES_DIR/*index.yaml
+      - id: upload-artifact
+        name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: indexes
+          path: ./${{ env.INDEXES_DIR }}
+          retention-days: 2
+          if-no-files-found: error
+  merge:
+    runs-on: ubuntu-latest
+    needs: get
+    name: Merge
+    steps:
+      - id: download-artifact
+        name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: indexes
+      - id: merge
+        name: Merge
+        run: yq eval-all '. as $item ireduce ({}; . *+ $item )' old_index.yaml last_index.yaml > duplicates_index.yaml
+      - id: upload-artifact
+        name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: duplicates-index
+          path: duplicates_index.yaml
+          retention-days: 2
+          if-no-files-found: error
+  deduplicate:
+    runs-on: ubuntu-latest
+    needs: merge
+    name: Deduplicate
+    steps:
+      - id: download-artifact
+        name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: duplicates-index
+      - id: remove
+        name: Remove duplicates
+        # Removes duplicates per entry using 'digest' as value.
+        run: yq eval '.entries[] |= unique_by(.digest)' duplicates_index.yaml > index.yaml
+      - id: upload-artifact
+        name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: archive-full-index
+          path: index.yaml
+          retention-days: 2
+          if-no-files-found: error
+  checks:
+    runs-on: ubuntu-latest
+    needs: deduplicate
+    name: Checks
+    steps:
+      - id: download-artifact-indexes
+        name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: indexes
+      - id: download-artifact-archive-full-index
+        name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: archive-full-index
```

Currently, we are uploading 3 intermediate artifacts: `indexes`, `duplicates-index`, and `archive-full-index`.
If we get rid of the intermediate `duplicates-index`, only two artifacts will be present: `indexes` and `archive-full-index`, which are the ones downloaded in this step. In that case, we can simplify those steps into a single one, since if no `name` is provided, all the present artifacts will be downloaded, see https://github.com/actions/download-artifact#download-all-artifacts

```suggestion
      - id: download-artifacts
        name: Download artifacts
        uses: actions/download-artifact@v3
```

---

I like it! I will get rid of the intermediate `duplicates-index`.

---

done!